src/diskquota_utility.c

```diff
@@ -1683,3 +1683,25 @@ DiskquotaShmemInitHash(const char           *name,       /* table string name fo
 	return ShmemInitHash(name, init_size, max_size, infoP, hash_flags | HASH_BLOBS);
 #endif /* GP_VERSION_NUM */
 }
+
+/*
+ * Returns HASH_FIND if hash table is full and HASH_ENTER otherwise.
+ */
+HASHACTION
+check_hash_fullness(HTAB *hashp, int max_size, const char *warning_message, TimestampTz *last_overflow_report,
```

Comment:
Can this function have a prototype like this?
```suggestion
check_hash_fullness(HTAB *hashp, int max_size, TimestampTz *last_overflow_report, const char *warning_message, ...)
```

Reply:
we cannot pass variadic arguments to another function. But it may be worth formatting the string once during initialization, since changing the GUC requires a restart. what do you think?

Reply:
> changing the GUC requires a restart

all used GUCs?

Reply:
I came up with a better solution! See below.